src/redux/cart/cart.reducer.js

```diff
@@ -0,0 +1,172 @@
+    //   sales: 1,
+    //   sold_out: true,
+    //   user_id: 19,
+    // },
+    // {
+    //   checked: false,
+    //   id: 8,
+    //   image_url:
+    //     "https://images.unsplash.com/photo-1598806243937-2072d39bc11d?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=633&q=80",
+    //   name: "간편 간식 브리또 8종",
+    //   option: 2,
+    //   option_name: "불고기 브리또",
+    //   option_price: 2800,
+    //   option_sales: 1,
+    //   option_sold_out: false,
+    //   price: 2800,
+    //   product_id: 5,
+    //   quantity: 1,
+    //   sales: 1,
+    //   sold_out: true,
+    //   user_id: 19,
+    // },
+    // {
+    //   checked: false,
+    //   id: 9,
+    //   image_url:
+    //     "https://images.unsplash.com/photo-1598806243937-2072d39bc11d?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=633&q=80",
+    //   name: "간편 간식 브리또 8종",
+    //   option: 3,
+    //   option_name: "야채 브리또",
+    //   option_price: 2800,
+    //   option_sales: 1,
+    //   option_sold_out: false,
+    //   price: 2800,
+    //   product_id: 5,
+    //   quantity: 1,
+    //   sales: 1,
+    //   sold_out: false,
+    //   user_id: 19,
+    // },
+  ],
+  allSelect: true,
+  seletectedItemsAmount: 0,
+  selectedItemsTotalPrice: 0,
+};
+
+const cartReducer = (state = INITIAL_STATE, action) => {
+  switch (action.type) {
+    case CartActionTypes.INCREASE_ITEM_AMOUNT:
+      return {
+        ...state,
+        cartItems: increaseItemAmount(state.cartItems, action.payload),
+      };
+    case CartActionTypes.DECREASE_ITEM_AMOUNT:
+      return {
+        ...state,
+        cartItems: removeItemFromCart(state.cartItems, action.payload),
+      };
+    case CartActionTypes.CLEAR_ITEM_FROM_CART:
+      return {
+        ...state,
+        cartItems: state.cartItems.filter(
+          (cartItem) => cartItem.id !== action.payload.id
+        ),
+      };
+    case CartActionTypes.TOGGLE_ITEM_CHECKBOX:
+      return {
+        ...state,
+        cartItems: toggleItemCheckBox(state.cartItems, action.payload),
+      };
+    case CartActionTypes.TOGGLE_ALL_SELECT_CHECKBOX:
+      return {
+        ...state,
+        allSelect: !state.allSelect,
+        cartItems: state.cartItems.map((cartItem) => {
+          cartItem.checked = !state.allSelect;
+          return cartItem;
+        }),
+      };
+    case CartActionTypes.CHECK_ALL_SELECT_CHECKBOX:
+      return {
+        ...state,
+        allSelect: checkAllSelectCheckBox(state.cartItems),
+      };
+    case CartActionTypes.DELETE_SELECTED_ITEMS:
+      return {
+        ...state,
+        cartItems: filterOutSelectedItems(state.cartItems),
+      };
+    case CartActionTypes.SELECTED_ITEMS_TOTAL_PRICE:
+      return {
+        ...state,
+        selectedItemsTotalPrice: state.cartItems.reduce(
+          (accumulator, cartItem) =>
+            cartItem.checked
+              ? accumulator + cartItem.price * cartItem.quantity
+              : accumulator,
+          0
+        ),
```

Comment:
```suggestion
        selectedItemsTotalPrice: state.cartItems.reduce(
          (accumulator, { checked, price, quantity }) =>
            checked ? accumulator + price * quantity : accumulator, 0
          ),
```
함수 파라미터 자리에서 직접 구조분해할당 할 수 있습니다

Reply:
수정했습니다! 😄